---
float64[] trajx
float64[] trajy
float64[] trajdx
float64[] trajdy